gems/smithy/lib/smithy/anvil/templates/client/plugins/endpoint.erb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+# This is generated code!
+
+module <%= namespace %>
+  module Plugins
+    class Endpoint < Smithy::Client::Plugin
+      option(
+        :endpoint_provider,
+        doc_type: '<%= namespace %>::EndpointProvider',
+        docstring: <<~DOCS) do |_cfg|
+The endpoint provider used to resolve endpoints. Any object that responds to
+`#resolve_endpoint(parameters)` where `parameters` is a Struct similar to
+`Aws::ACM::EndpointParameters`.
+        DOCS
+        EndpointProvider.new
+      end
+
+      <%- parameters.select(&:built_in?).each do |param| -%>
+      <%= param.built_in_binding.render_config(plan) %>
```

Comment:
Wouldn't param already know about plan? You can probably simplify this to just have a method that returns configs to render. In general we can have logic in these but I think it makes sense to minimize that

Reply:
Param knows about plan but the built in binding does not and needs to be called with it.  I could make a convience method on the `EndpointParameter` view helper for `render_builtin_binding_config` which wouldn't need to pass plan (since the parameter knows it), but that doesn't seem much better.

The reason I went with a generic block was to keep things simple - I didn't want to model a codegen config the way we do in smithy-ruby java.  The configs can be fairly complex (eg, they need some arbitrary default or resolution logic).